lib/_http_agent.js

```diff
@@ -88,11 +90,22 @@ function Agent(options) {
   this.maxSockets = this.options.maxSockets || Agent.defaultMaxSockets;
   this.maxFreeSockets = this.options.maxFreeSockets || 256;
   this.scheduling = this.options.scheduling || 'fifo';
+  this.maxTotalSockets = this.options.maxTotalSockets;
+  this.totalSocketCount = 0;
 
   if (this.scheduling !== 'fifo' && this.scheduling !== 'lifo') {
     throw new ERR_INVALID_OPT_VALUE('scheduling', this.scheduling);
   }
 
+  if (this.maxTotalSockets !== undefined) {
+    validateNumber(this.maxTotalSockets, 'maxTotalSockets');
+    if (this.maxTotalSockets <= 0)
```

Comment:
`NaN` will pass these two checks, so we should probably explicitly check for it as well.

Reply:
Add an integer check.